l1-contracts/test/TestERC20.t.sol

```diff
@@ -1,24 +1,90 @@
-pragma solidity ^0.8.18;
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.27;
 
-import "forge-std/Test.sol";
+import {Test} from "forge-std/Test.sol";
 import {TestERC20} from "@aztec/mock/TestERC20.sol";
 
+// solhint-disable comprehensive-interface
+// solhint-disable func-name-mixedcase
+
 contract TestERC20Test is Test {
-  TestERC20 testERC20;
+  TestERC20 internal testERC20;
 
   function setUp() public {
     testERC20 = new TestERC20("test", "TEST", address(this));
   }
 
+  function test_addRemoveMinter(address _minter) public {
+    vm.assume(_minter != address(this));
+    vm.assume(_minter != address(0));
+
+    vm.prank(address(_minter));
+    vm.expectRevert();
+    testERC20.mint(address(this), 100);
+
+    vm.expectEmit();
+    emit TestERC20.MinterAdded(_minter);
+    testERC20.addMinter(address(_minter));
+
+    vm.prank(address(_minter));
+    testERC20.mint(address(this), 100);
+
+    vm.expectEmit();
+    emit TestERC20.MinterRemoved(_minter);
+    testERC20.removeMinter(address(_minter));
+
+    vm.prank(address(_minter));
+    vm.expectRevert();
+    testERC20.mint(address(this), 100);
+  }
+
+  function test_onlyOwnerCanAddRemoveMinter(address _minter, address _caller) public {
+    vm.assume(_minter != address(this));
+    vm.assume(_minter != address(0));
+
+    vm.assume(_caller != address(_minter));
```

Comment:
Not sure about these as well, only the owner should be able to add, so we don't need to constrain on these as well.